src/store/RelayStoreData.js

```diff
@@ -632,6 +633,14 @@ class RelayStoreData {
       nodeRangeMap,
     } = obj;
 
+    for (const key in records) {
+      const record = records[key];
+      const range = record.__range__;
+      if (range) {
+        record.__range__ = GraphQLRange.fromJSON(range);
+      }
+    }
+
```

Comment:
I should probably break this logic out into a utility then, where would something like that live in the codebase?